python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/run_handler.py

```diff
@@ -195,7 +196,25 @@ def to_default_asset_events(
                     test_unique_id=unique_id,
                 )
 
-                if asset_check_key is not None:
+                if (
+                    context
+                    and has_asset_def
+                    and asset_check_key is not None
+                    and asset_check_key in context.selected_asset_check_keys
+                ):
+                    # The test is an asset check in an asset, so yield an `AssetCheckResult`.
+                    yield AssetCheckResult(
+                        passed=result_status == TestStatus.Pass,
+                        asset_key=asset_check_key.asset_key,
+                        check_name=asset_check_key.name,
+                        metadata=metadata,
+                        severity=(
+                            AssetCheckSeverity.WARN
+                            if result_status == TestStatus.Warn
+                            else AssetCheckSeverity.ERROR
+                        ),
+                    )
+                elif asset_check_key is not None:
```

Comment:
And this means that it's an asset check in either direct invocation or that the check wasn't selected? Does this mirror DBT behavior?

Reply:
> And this means that it's an asset check in either direct invocation or that the check wasn't selected?

I just double checked and I think the condition should be `elif not has_asset_def and asset_check_key is not None` instead. We should not yield an evaluation in an asset context if the check was not selected. This would also reflect the behavior in dbt core.

In dbt core, we use asset observations for tests that are executed but not selected. I agree that we should share the code between dbt core and dbt Cloud here to avoid conflicts and behavior discrepancy. I will update the condition for this PR, and reuse the same code for both integrations in another PR.